packages/boot/test/bootstrapTests/ec-contract-upgrade.test.ts

```diff
@@ -0,0 +1,601 @@
+
+export const makeZoeTestContext = async t => {
+  console.time('ZoeTestContext');
+  const swingsetTestKit = await makeSwingsetTestKit(t.log, undefined, {
+    configSpecifier: '@agoric/vm-config/decentral-main-vaults-config.json',
+  });
+
+  const { runUtils, storage } = swingsetTestKit;
+  console.timeLog('DefaultTestContext', 'swingsetTestKit');
+  const { EV } = runUtils;
+
+  await eventLoopIteration();
+
+  // We don't need vaults, but this gets the brand, which is checked somewhere
+  // Wait for ATOM to make it into agoricNames
+  await EV.vat('bootstrap').consumeItem('vaultFactoryKit');
+  console.timeLog('DefaultTestContext', 'vaultFactoryKit');
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(storage);
+  console.timeLog('DefaultTestContext', 'agoricNamesRemotes');
+
+  console.timeEnd('DefaultTestContext');
+  const walletFactoryDriver = await makeWalletFactoryDriver(
+    runUtils,
+    storage,
+    agoricNamesRemotes,
+  );
+
+  const smartWallets = await Promise.all(
+    Object.values(wallets).map(async addr =>
+      walletFactoryDriver.provideSmartWallet(addr),
+    ),
+  );
+  const { fromCapData } = makeMarshal(undefined, slotToBoardRemote);
+
+  return {
+    ...swingsetTestKit,
+    storage,
+    smartWallets,
+    fromCapData,
+  };
+};
+const test = anyTest as TestFn<Awaited<ReturnType<typeof makeZoeTestContext>>>;
+
+test.before(async t => {
+  t.context = await makeZoeTestContext(t);
+});
+
+test.serial('normal running of committee', async t => {
+  const { advanceTimeTo, storage, smartWallets, fromCapData } = t.context;
+
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(storage);
+  const { econCommitteeCharter, economicCommittee, VaultFactory } =
+    agoricNamesRemotes.instance;
+  const { ATOM: collateralBrand, IST: debtBrand } = agoricNamesRemotes.brand;
+
+  // Accepting all invitations for original committee
+  await Promise.all(
+    smartWallets.map(w =>
+      w.executeOffer({
+        id: PROPOSAL_INV,
+        invitationSpec: {
+          source: 'purse',
+          instance: econCommitteeCharter,
+          description: 'charter member invitation',
+        },
+        proposal: {},
+      }),
+    ),
+  );
+
+  await Promise.all(
+    smartWallets.map(w => {
+      const invitationPurse = w.getCurrentWalletRecord().purses.find(p => {
+        return p.brand.toString().includes('Invitation');
+      });
+      if (!invitationPurse) return null;
+
+      // @ts-ignore
+      const invitation = invitationPurse.balance.value.find(a =>
+        a.description.includes('Voter'),
+      );
+      return w.executeOffer({
+        id: VOTER_INV,
+        invitationSpec: {
+          source: 'purse',
+          instance: economicCommittee,
+          description: invitation.description,
+        },
+        proposal: {},
+      });
+    }),
+  );
+
+  // Proposing a question using first wallet
```

Comment:
Consider using `t.log()` so that the output tells the user story.

```suggestion
  t.log('Proposing a question using first wallet');
```